diff -ruN CacheLib-e3703aade03d359d290936b334ab81ca4a856b41/cachelib/common/CMakeLists.txt CacheLib-ld_gtest/cachelib/common/CMakeLists.txt
--- CacheLib-e3703aade03d359d290936b334ab81ca4a856b41/cachelib/common/CMakeLists.txt	2021-11-29 13:01:49.000000000 -0800
+++ CacheLib-ld_gtest/cachelib/common/CMakeLists.txt	2021-12-22 10:21:59.259653149 -0800
@@ -21,15 +21,11 @@
   CountDownLatch.cpp
   ${BLOOM_THRIFT_FILES}
   hothash/HotHashDetector.cpp
-  hothash/HotHashDetectorTest.cpp
   PercentileStats.cpp
   PeriodicWorker.cpp
   piecewise/GenericPieces.cpp
-  piecewise/GenericPiecesTest.cpp
   piecewise/RequestRange.cpp
-  piecewise/RequestRangeTest.cpp
   Serialization.cpp
-  TestUtils.cpp
   Utils.cpp
 )
 add_dependencies(cachelib_common thrift_generated_files)
@@ -51,7 +47,12 @@
 
 
 if (BUILD_TESTS)
-  add_library (common_test_support INTERFACE)
+  add_library (common_test_support INTERFACE
+    TestUtils.cpp
+    hothash/HotHashDetectorTest.cpp
+    piecewise/GenericPiecesTest.cpp
+    piecewise/RequestRangeTest.cpp
+  )
   target_link_libraries (common_test_support INTERFACE
     cachelib_common
     gflags
