diff -ruN CacheLib-e3703aade03d359d290936b334ab81ca4a856b41/cachelib/CMakeLists.txt CacheLib-versioned_so/cachelib/CMakeLists.txt
--- CacheLib-e3703aade03d359d290936b334ab81ca4a856b41/cachelib/CMakeLists.txt	2021-11-29 13:01:49.000000000 -0800
+++ CacheLib-versioned_so/cachelib/CMakeLists.txt	2021-12-22 14:22:35.444840967 -0800
@@ -24,14 +24,18 @@
 
 #configure_file(cachelib/cachelib_config.h.in cachelib_config.h)
 
-set(CACHELIB_MAJOR_VERSION 0)
+if (NOT DEFINED CACHELIB_MAJOR_VERSION)
+  set(CACHELIB_MAJOR_VERSION 0)
+endif ()
 set(CACHELIB_MINOR_VERSION 1)
 set(CACHELIB_PATCH_VERSION 0)
 set(CACHELIB_VERSION
  ${CACHELIB_MAJOR_VERSION}.${CACHELIB_MINOR_VERSION}.${CACHELIB_PATCH_VERSION})
 
 set(PACKAGE_NAME      "cachelib")
-set(PACKAGE_VERSION   "${CACHELIB_VERSION}")
+if (NOT DEFINED PACKAGE_VERSION)
+  set(PACKAGE_VERSION   "${CACHELIB_VERSION}")
+endif ()
 set(PACKAGE_STRING    "${PACKAGE_NAME} ${PACKAGE_VERSION}")
 set(PACKAGE_TARNAME   "${PACKAGE_NAME}-${PACKAGE_VERSION}")
 set(PACKAGE_BUGREPORT "https://github.com/facebook/TBD")
@@ -361,6 +365,20 @@
         #NAMESPACE cachelib::
         DESTINATION ${CMAKE_INSTALL_DIR})
 
+if (BUILD_SHARED_LIBS)
+  set_target_properties(
+    cachelib_allocator
+    cachelib_cachebench
+    cachelib_common
+    cachelib_datatype
+    cachelib_navy
+    cachelib_shm
+    PROPERTIES
+    SOVERSION ${CACHELIB_MAJOR_VERSION}
+    VERSION ${PACKAGE_VERSION}
+  )
+endif ()
+
 if (BUILD_TESTS)
   get_property(TEST_BINARIES GLOBAL PROPERTY TEST_BINARIES)
   #message(STATUS "=== Test binaries : ${TEST_BINARIES} ===")
