--- a/cachelib/cachebench/cache/ItemRecords.h
+++ b/cachelib/cachebench/cache/ItemRecords.h
@@ -102,11 +102,11 @@ class ItemRecords {
     }
     if (record.destructCount != 0) {
       XLOGF(ERR, "unexpected destructCount {} for item {}, context {}|{}",
-            record.destructCount, item.getKey(), data.context, record.context);
+            record.destructCount, item.getKey().toString(), data.context, record.context);
     }
     if (record.version != ptr->getVersion()) {
       XLOGF(ERR, "unexpected version {}|{} for item {}", record.version,
-            ptr->getVersion(), item.getKey());
+            ptr->getVersion(), item.getKey().toString());
     }
     ++record.destructCount;
 
